NOTICE

```diff
@@ -17,9 +17,42 @@ under the auspices of the Eigenbase Foundation
 and released as the LucidDB project.
 
 
+================= Apache Flink =================
+Apache Flink
+Copyright 2014-2019 The Apache Software Foundation
 
 
-================= Metamarkets extendedset 1.3.10 =================
+================= Apache Hadoop =================
+Apache Hadoop
+Copyright 2009-2017 The Apache Software Foundation
+
+
+================= Apache Hive =================
```

Comment:
https://github.com/apache/incubator-druid/pull/3018 adapted from [Hive 2.0](https://github.com/apache/hive/blob/branch-2.0/storage-api/src/java/org/apache/hadoop/hive/ql/io/sarg/SearchArgumentImpl.java).

Reply:
https://github.com/apache/incubator-druid/pull/2525 was raised on Feb 23, 2016 and [Hive 2.0.1 was released on May 24, 2016](https://github.com/apache/hive/releases?after=rel%2Fstorage-release-2.4.0). The version would be 2.0.1.

Reply:
[BloomKFilter was adapted from Hive 2.7.0](https://github.com/apache/incubator-druid/blob/master/extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/filter/BloomKFilter.java#L36).